wpilibj/src/main/java/edu/wpi/first/wpilibj/DriverStation.java

```diff
@@ -391,26 +390,26 @@ public double getStickAxis(int stick, int axis) {
     if (stick < 0 || stick >= kJoystickPorts) {
       throw new RuntimeException("Joystick index is out of range, should be 0-5");
     }
-    if (axis < 0 || axis >= HAL.kMaxJoystickAxes) {
-      throw new RuntimeException("Joystick axis is out of range");
-    }
-
-    boolean error = false;
-    double retVal = 0.0;
-    synchronized (m_cacheDataMutex) {
+    m_cacheDataMutex.lock();
+    try {
       if (axis >= m_joystickAxes[stick].m_count) {
-        // set error
-        error = true;
-        retVal = 0.0;
-      } else {
-        retVal =  m_joystickAxes[stick].m_axes[axis];
+        // Unlock early so error printing isn't locked.
+        m_cacheDataMutex.unlock();
+        if (axis >= HAL.kMaxJoystickAxes) {
+          throw new RuntimeException("Joystick axis is out of range");
```

Comment:
Shouldn't this argument check stay at the top, and continue ensuring it's non-negative? (I guess the C++ one should also be fixed.)